### PR TITLE
Add pipeline metadata to art imagestreams

### PR DESCRIPTION
**Pull request by jupierce**

### CODE REVIEW

**Comment by jupierce** [comment]
Aligns with https://github.com/openshift/release-controller/pull/461 to propagate the annotation into the nightlies.

File: doozerlib/cli/release_gen_payload.py [hunk 1]
```diff
@@ -1210,14 +1225,14 @@ def _find_rhcos_payload_entries(assembly_inspector: AssemblyInspector, arch: str
         return members, issues
 
     @staticmethod
-    def build_payload_istag(payload_tag_name: str, payload_entry: PayloadEntry) -> Dict:
+    def build_payload_istag(runtime, payload_tag_name: str, payload_entry: PayloadEntry) -> Dict:
```
Comment (vfreex) [inline doozerlib/cli/release_gen_payload.py:1228]: `runtime` is not used.

```suggestion
    def build_payload_istag(payload_tag_name: str, payload_entry: PayloadEntry) -> Dict:
```

File: doozerlib/cli/release_gen_payload.py [hunk 2]
```diff
@@ -879,6 +882,7 @@ def build_multi_istag(self, tag_name: str, arch_to_payload_entry: Dict[str, Payl
                       for payload_entry in entries
                       for issue in payload_entry.issues or [])
         return PayloadGenerator.build_payload_istag(
+            self.runtime,
```
Comment (vfreex) [inline doozerlib/cli/release_gen_payload.py:885]: `runtime` is not used.

```suggestion
```

**Review by vfreex — CHANGES_REQUESTED** [review]
One minor issue, and unit test needs a fix. Otherwise LGTM.

File: doozerlib/cli/release_gen_payload.py [hunk 3]
```diff
@@ -1226,7 +1241,7 @@ def build_payload_istag(payload_tag_name: str, payload_entry: PayloadEntry) -> D
         }
 
     @staticmethod
-    def build_payload_imagestream(imagestream_name: str, imagestream_namespace: str, payload_istags: Iterable[Dict], assembly_wide_inconsistencies: Iterable[AssemblyIssue]) -> Dict:
+    def build_payload_imagestream(runtime, imagestream_name: str, imagestream_namespace: str, payload_istags: Iterable[Dict], assembly_wide_inconsistencies: Iterable[AssemblyIssue]) -> Dict:
```
Comment (locriandev) [inline doozerlib/cli/release_gen_payload.py:1244]: Docstring should include `runtime` param description

Comment (jupierce) [reply doozerlib/cli/release_gen_payload.py:1243]: Added

File: doozerlib/cli/release_gen_payload.py (hunk 2 above)
Comment (jupierce) [reply doozerlib/cli/release_gen_payload.py]: Updated

File: doozerlib/cli/release_gen_payload.py (hunk 1 above)
Comment (jupierce) [reply doozerlib/cli/release_gen_payload.py]: Updated

**Review by vfreex — APPROVED** [review]
/lgtm

**Pull request merged**